File: restaurant/model/menu_item.py
from model.enums import DishStatus

class MenuItem:
    """Model đại diện cho một mục trong thực đơn"""
    
    def __init__(self, item_id: int, name: str, price: float, description: str = ""):
        self.item_id = item_id
        self.name = name
        self.price = price
        self.description = description
        self.status = DishStatus.AVAILABLE
    
    def set_available(self):
        """Đánh dấu món ăn là có sẵn"""
        self.status = DishStatus.AVAILABLE
    
    def set_unavailable(self):
        """Đánh dấu món ăn là không có sẵn"""
        self.status = DishStatus.NOT_AVAILABLE
    
    def __repr__(self):
        return f"MenuItem(id={self.item_id}, name={self.name}, price={self.price}, status={self.status.value})"

File: backend/src/com/restaurant/model/entity/MenuItem.java
package com.restaurant.model.entity;

import com.restaurant.model.enums.MenuItemStatus;
import java.time.LocalDateTime;

/**
 * Entity đại diện cho một mục trong thực đơn
 */
public class MenuItem {
    private final int itemId;
    private final String name;
    private final double price;
    private final String description;
    private MenuItemStatus status;
    private final LocalDateTime createdAt;

    public MenuItem(int itemId, String name, double price, String description) {
        this.itemId = itemId;
        this.name = name;
        this.price = price;
        this.description = description;
        this.status = MenuItemStatus.AVAILABLE;
        this.createdAt = LocalDateTime.now();
    }

    public MenuItem(int itemId, String name, double price) {
        this(itemId, name, price, "");
    }

    // Getters
    public int getItemId() {
        return itemId;
    }

    public String getName() {
        return name;
    }

    public double getPrice() {
        return price;
    }

    public String getDescription() {
        return description;
    }

    public MenuItemStatus getStatus() {
        return status;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    // Business Methods
    public void setAvailable() {
        this.status = MenuItemStatus.AVAILABLE;
    }

    public void setOutOfStock() {
        this.status = MenuItemStatus.OUT_OF_STOCK;
    }

    public boolean isAvailable() {
        return status == MenuItemStatus.AVAILABLE;
    }

    @Override
    public String toString() {
        return String.format("%s - %,.0fđ", name, price);
    }
}
